src/main/java/core/basesyntax/service/shop/ShopServiceImpl.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.service.shop;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.FruitTransactionStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final FruitTransactionStrategy fruitTransactionStrategy;
+
+    public ShopServiceImpl(FruitTransactionStrategy fruitTransactionStrategy) {
+        this.fruitTransactionStrategy = fruitTransactionStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        for (FruitTransaction f : fruitTransactions) {

```

The loop assumes every `FruitTransaction f` and `f.getOperation()` is non-null. You should validate each element before use and throw informative RuntimeException for invalid data. This also relates to the checklist requirement: "Null parameters should be handled (throw informative RuntimeException)". Consider checking `f` and `f.getOperation()` and including the transaction/index in the error message.